### ISSUE TITLE
Memory leak in cartesianBase

### ISSUE THREAD

**Issue by ColinEberhardt** [d3fc/d3fc#1152]
_From @Lauri-Nomme on May 9, 2018 16:6_

In https://github.com/d3fc/d3fc-chart/blob/master/src/cartesianBase.js#L127, `auto-resize` attribute is set on `container` enter selection, which contains a `d3fc-group` element.
This causes a resize eventListener to be added to the global window in https://github.com/d3fc/d3fc-element/blob/master/src/group.js#L32.
The attribute is never set to false (there is no mechanism to 'destruct' the chart).

As the handler's closure has a reference to `this`, the element never gets GCd, causing a leak of all the data set on the chart + the child svg nodes etc.

_Copied from original issue: d3fc/d3fc-chart#37_

**Comment by ColinEberhardt** [comment]
_From @msuperina on June 27, 2018 9:36_

Would it be sufficient to do
```
updateAutoResize() {
        if (this.autoResize) {
            if (this.__autoResizeListener__ == null) {
                this.__autoResizeListener__ = () => requestRedraw(this);
            } else {
                // **** change here ****
                // remove any previously added event listener 
                removeEventListener('resize', this.__autoResizeListener__);
            }
            addEventListener('resize', this.__autoResizeListener__);
        } else {
            removeEventListener('resize', this.__autoResizeListener__);
        }
}
```
at https://github.com/d3fc/d3fc-element/blob/master/src/group.js#L27 ?

**Comment by chrisprice** [comment]
Fixed by #1179
